sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -794,4 +795,28 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
       }
     }
   }
+
+  private def checkIfNoJobTriggered(f: => DataFrame): DataFrame = {
+    var numJobTrigered = 0
+    val jobListener = new SparkListener {
+      override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+        numJobTrigered += 1
+      }
+    }
+    sparkContext.addSparkListener(jobListener)
+    try {
+      val df = f
+      assert(numJobTrigered === 0)
+      df
+    } finally {
+      sparkContext.removeSparkListener(jobListener)
+    }
+  }
+
+  test("SPARK-23880 table cache should be lazy and don't trigger any jobs") {
```

Comment:
Without the changes in this PR, this test still can pass. : )

Reply:
oh, I'll recheck. Thanks!